src/app/public/modules/modal-provider/types/modal-provider.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  EventEmitter
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
+
+import {
+  SkyModalProviderCloseArgs
+} from './modal-provider-close-args';
+
+export interface SkyModalProvider {
+
+  type: string;
+  closeCallback: Observable<SkyModalProviderCloseArgs>;
```

Comment:
When I think of a "callback" I think of a function. We might be okay with just `close`. If the observable unsubscribes automatically after emitting, it should be in the past tense, `closed`.

Reply:
I changed this to `closed`